packages/create/src/cli/change/change-data-source.ts

```diff
@@ -0,0 +1,133 @@
+import { getLogger, traceChanges } from '../../tracing';
+import { readFileSync, existsSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { promptYUIQuestions } from '../../common';
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - If set to true, then no files will be written to the filesystem.
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        checkEnvironment(basePath);
+
+        const variant = getVariant(basePath);
+        const manifest = await getManifest(basePath, logger, variant);
+        const dataSources = manifest['sap.app'].dataSources;
+        if (!dataSources) {
+            throw new Error('No data sources found in the manifest');
+        }
+        const answers = await promptYUIQuestions(getPromptsForChangeDataSource(dataSources), false);
+
+        const fs = await generateChange<ChangeType.CHANGE_DATA_SOURCE>(basePath, ChangeType.CHANGE_DATA_SOURCE, {
+            variant,
+            dataSources,
+            answers
+        });
+
+        if (!simulate) {
+            await new Promise((resolve) => fs.commit(resolve));
+        } else {
+            await traceChanges(fs);
+        }
+    } catch (error) {
+        logger.error(error.message);
+        if (error.response?.status === 401 && loginAttempts) {
+            loginAttempts--;
+            logger.error(`Authentication failed. Please check your credentials. Login attempts left: ${loginAttempts}`);
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        logger.debug(error);
+    }
+}
+
+/**
+ * Get the manifest of the base application.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {ToolsLogger} logger - The logger.
+ * @param {DescriptorVariant} variant - The app descriptor variant.
+ * @returns {Promise<Manifest>} The manifest.
+ */
+async function getManifest(basePath: string, logger: ToolsLogger, variant: DescriptorVariant): Promise<Manifest> {
```

Comment:
This method will be reused in other "generators", so it should be moved to a some common file?

Reply:
- The Sonar issues and code coverage are currently coming from a functionality that @tobiasqueck provided for us. This was originally part of its own PR (https://github.com/SAP/open-ux-tools/pull/1940) . This point needs to be discussed with Tobias to confirm whether those changes would be part of this PR or the other.
- Regarding the rosolving of the comments, I followed the documentation here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#resolving-conversations. It suggests that owner of the PR resolves the comments, making it easier to find conversations that still need to be address. We could discuss if following this process is not prefered.
- Currently, there is a common method `getManifest` in `@sap-ux/axios-extension`, which can be reused in other generator. The logic in the `getManifest` method you point to is more specific to this generator. But if it is the case that this logic can be reused when other generators are migrated, it can be refactored then.

Reply:
Every organization, and repository has its own set of rules, for open-ux-tools it is the following: The reviewer resolves his comments which is logical. This is like a conversation, I ask something and you answer. But who should say whether the answer is sufficient, acceptable, or correct? I think the reviewer since he is the one who asked. If you decide to resolve the comment yourself that means first the comment is hidden now, and when I go to check your answer I can't find it and I have to go through all the code to check whether it was resolved. Second, if you replied you automatically think this is a correct answer, but this is a conversation, and I may disagree with your answer. I hope that clears it out. :)

If you received approval but I for example forgot to resolve some of my comments, then you can resolve them yourself.

Reply:
Yes, this made it clear :). @testojs made me aware that this is mentioned in [open-ux-tools/CONTRIBUTING.md at 59410cffdc3de151c0c35aa445a586c56e62f321 · SAP/open-ux-tools (github.com)](https://github.com/SAP/open-ux-tools/blob/59410cffdc3de151c0c35aa445a586c56e62f321/CONTRIBUTING.md#how-to-contribute). I will have it in mind for future comments. Btw, you could expand all resolved comments in a PR by Alt + Click / Option + Click on one of them. Hopefully, this will make it easier to find my answers in this PR

Reply:
For no leaving this comment open TBD with Tobias, since I do not understand why those Sonar and tests do not need to be written for the functionality he provided to you.

Reply:
@nikmace there is nothing to understand, the tests are just missing - need to do them. I was working on the functionality in a separate branch ensuring that the approach works for two different ongoing implementations. Both work with it now, so next will be the tests, and then it is either a separate PR or part of this one (currently I prefer separate.

Reply:
@nikmace the quality of that code is now as required